File: symfony/src/Service/FrontTestService.php
<?php

namespace App\Service;

class FrontTestService
{
    public function determineCategory(int $score): string
    {
        if ($score <= 10) {
            return 'Stress';
        }

        if ($score <= 20) {
            return 'Anxiete';
        }

        if ($score <= 30) {
            return 'Depression';
        }

        return 'Trouble du Sommeil';
    }

    public function determineLevel(int $percentage): string
    {
        if ($percentage < 34) {
            return 'Faible';
        }

        if ($percentage < 67) {
            return 'Moyen';
        }

        return 'Élevé';
    }

    public function calculatePercentage(int $score, int $maxScore): int
    {
        if ($maxScore <= 0) {
            return 0;
        }

        return (int) round(($score / $maxScore) * 100);
    }
}

File: integrated-project/mindboost-javafx/src/main/java/com/mindboost/app/service/FrontTestService.java
package com.mindboost.app.service;

public class FrontTestService {
    public String determineCategory(int score) {
        if (score <= 10) {
            return "Stress";
        }
        if (score <= 20) {
            return "Anxiete";
        }
        if (score <= 30) {
            return "Depression";
        }
        return "Trouble du Sommeil";
    }

    public String determineLevel(int percentage) {
        if (percentage < 34) {
            return "Faible";
        }
        if (percentage < 67) {
            return "Moyen";
        }
        return "Élevé";
    }

    public int calculatePercentage(int score, int maxScore) {
        if (maxScore <= 0) {
            return 0;
        }
        return (int) Math.round((score / (double) maxScore) * 100);
    }
}
